wadas_webserver/database.py

```diff
@@ -0,0 +1,122 @@
+import logging
+from typing import List, Optional, Tuple
+
+from sqlalchemy import and_, create_engine
+from sqlalchemy.orm import sessionmaker
+
+from wadas.domain.db_model import Camera as DB_Camera
+from wadas.domain.db_model import ClassifiedAnimals as DB_ClassifiedAnimal
+from wadas.domain.db_model import DetectionEvent as DB_DetectionEvent
+from wadas.domain.db_model import User as DB_User
+from wadas_webserver.mapper import Mapper
+from wadas_webserver.view_model import Camera, DetectionEvent, User
+
+logger = logging.getLogger(__name__)
+
+
+class Database:
+    instance = None
+
+    def __init__(self, connection_string):
+        self.connection_string = connection_string
+        self.engine = create_engine(self.get_connection_string())
+
+    def get_connection_string(self):
+        return self.connection_string
+
+    def get_user_by_username(self, username: str) -> Optional[User]:
+        """Method to get a user object from his username"""
+        session = sessionmaker(bind=self.engine)()
+        user = None
```

Comment:
As a general rule, variables (like "user", here. And below on all other cases) to be given back don't need to be initialized, because if something bad happens, then only the finally clause is executed and the below return statement will never be executed (e.g.: an exception is raised).